extensions/arc/src/test/ui/components/filePicker.test.ts

```diff
@@ -0,0 +1,70 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as should from 'should';
+import * as sinon from 'sinon';
+import * as vscode from 'vscode';
+import { Deferred } from '../../../common/promise';
+import { FilePicker } from '../../../ui/components/filePicker';
+import { createModelViewMock } from '../../stubs';
+
+let filePicker: FilePicker;
+const initialPath = '/path/to/.kube/config';
+const newFilePath = '/path/to/new/.kube/config';
+let filePathInputBox: azdata.InputBoxComponent;
+let browseButton: azdata.ButtonComponent;
+let flexContainer: azdata.FlexContainer;
+const browseButtonEmitter = new vscode.EventEmitter<undefined>();
+describe('filePicker', function (): void {
+	beforeEach(async () => {
+		const { mockModelBuilder, mockInputBoxBuilder, mockButtonBuilder, mockFlexBuilder } = createModelViewMock(browseButtonEmitter);
+		filePicker = new FilePicker(mockModelBuilder.object, initialPath, (_disposable) => { });
+		filePathInputBox = mockInputBoxBuilder.object.component();
+		browseButton = mockButtonBuilder.object.component();
+		flexContainer = mockFlexBuilder.object.component();
+	});
+
+	afterEach(() => {
+		sinon.restore();
+	});
+
+	it('browse Button chooses new FilePath', async () => {
+		should(filePathInputBox.value).should.not.be.undefined();
+		filePicker.value!.should.equal(initialPath);
+		flexContainer.items.should.deepEqual([filePathInputBox, browseButton]);
+		const deferred = new Deferred();
+		sinon.stub(vscode.window, 'showOpenDialog').callsFake((_options) =>
+			Promise.resolve([<vscode.Uri>{ fsPath: newFilePath }])
```

Comment:
Doesn't seem necessary to have the Promise.resolve and .then parts. This should do the same thing:

```
deferred.resolve();
return [vscode.Uri.file(newFilePath)];
```

Reply:
Done.